userspace/plugin/plugin_api.h

```diff
@@ -838,6 +838,15 @@ typedef struct {
 		//
 		const char* (*get_extract_event_sources)();
 
+		//
+		// Return this plugin's ability to extract field metadata. This
+		// will be invoked only once by the framework after the plugin's
+		// initialization.
+		//
+		// This function is optional. If NULL or 0, this indicates that
+		// the plugin does not support metadata extraction.
+		uint32_t (*get_extract_metadata_capability)();
+
```

Comment:
This will break the ABI for static plugins again :( I guess this is my life now (see https://github.com/falcosecurity/libs/pull/2315)

Also, what does this return on success? Anything non-zero? How about a ss_plugin_rc instead?

Reply:
It was supposed to return a bool; I used a uint32 because it appeared to be required for compatibility with the Go SDK.